Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -4,162 +4,142 @@
-                             "since no 'words' parameter was supplied in web request"
-                             });
-                         }
-                         catch (Exception)
-                         {
-                             outRecord.Errors.Add(new WebApiErrorWarningContract
-                             {
-                                 Message = "Could not parse predefined words.json"
-                             });
-                             return outRecord;
-                         }
-                     }
+                     IList<string> words = _userDefinedEntities.Words;
+                     Dictionary<string, string[]> synonyms = _userDefinedEntities.Synonyms;
+                     IList<string>  exactMatches = _userDefinedEntities.ExactMatch;
+                     int fuzzyEditDistance = Math.Max(0, _userDefinedEntities.FuzzyEditDistance);
+                     bool caseSensitive = _userDefinedEntities.CaseSensitive;
 
                      var entities = new List<Entity>();
                      var entitiesFound = new HashSet<string>();
+
                      if (!string.IsNullOrWhiteSpace(text))
                      {
-                         foreach (string word in words)
+                         // exact matches user regex
+                         if (fuzzyEditDistance == 0)
                          {
-                             if (string.IsNullOrEmpty(word)) continue;
-                             int leniency = (exactMatches != null && exactMatches.Contains(word)) ? 0 : offset;
-                             string wordCharArray = (caseSensitive) ? CreateWordArray(word) : CreateWordArray(word.ToLower(CultureInfo.CurrentCulture));
-                             if (leniency >= wordCharArray.Length)
+                             MatchCollection entityMatch = _precompiledExactMatchRegex.Matches(text);
+                             if (entityMatch.Count > 0)
                              {
-                                 outRecord.Warnings.Add(new WebApiErrorWarningContract
+                                 foreach (Match match in entityMatch)
                                  {
-                                     Message = @"The provided fuzzy offset of " + leniency + @", is larger than the length of the provided word, """ + word + @"""."
-                                 });
-                                 leniency = Math.Max(0, wordCharArray.Length - 1);
+                                     entities.Add(
+                                         new Entity
+                                         {
+                                             Category = "customEntity",
+                                             Value = match.Groups[1].Value,
+                                             Offset = match.Index,
+                                             Confidence = 0
```

Comment:
Seems weird to assign zero confidence to an exact match (but I'm assuming this is never used for exact matches?)

Reply:
sooo the "confidence" is actually the edit distance.. so 0 means there's no characters different. I didn't want to mess with the schema too much since they're using this skill in workshops this week

---
In reply to: [330762711](https://github.com/Azure-Samples/azure-search-power-skills/pull/14#discussion_r330762711) [](ancestors = 330762711)

Reply:
Oooh, right. OK. Yes, that's confusing. Should be renamed eventually.